modules/gdnative/gdnative/variant.cpp

```diff
@@ -158,7 +159,21 @@ void GDAPI godot_variant_new_rid(godot_variant *r_dest, const godot_rid *p_rid)
 void GDAPI godot_variant_new_object(godot_variant *r_dest, const godot_object *p_obj) {
 	Variant *dest = (Variant *)r_dest;
 	Object *obj = (Object *)p_obj;
-	memnew_placement_custom(dest, Variant, Variant(obj));
+	Reference *reference = Object::cast_to<Reference>(obj);
+	REF ref;
+	if (reference) {
```

Comment:
Can we really continue if cast_to() returns NULL here?

Reply:
Yes, it can be a non-Reference Object and it's fine.